tests/causal-inference-discovery-book/test_causal_inference_discovery_book.py

```diff
@@ -0,0 +1,189 @@
+                            "init_params": {
+                                "model_y": GradientBoostingRegressor(),
+                                "model_t": GradientBoostingRegressor(),
+                                "model_final": LassoCV(fit_intercept=False),
+                            },
+                            "fit_params": {},
+                        },
+                    )
+
+                    # Get estimate (Linear Regression)
+                    estimate_lr = model.estimate_effect(
+                        identified_estimand=estimand, method_name="backdoor.linear_regression"
+                    )
+
+                    results.append(
+                        {
+                            "sample_size": sample_size,
+                            "noise_coef": noise_coef,
+                            "estimate_dml": estimate_dml.value,
+                            "estimate_lr": estimate_lr.value,
+                            "error_dml": estimate_dml.value - TRUE_EFFECT,
+                            "error_lr": estimate_lr.value - TRUE_EFFECT,
+                        }
+                    )
+
+    def test_dowhy_chapter_7(self):
+        # Instantiate the SCM
+        scm = GPSMemorySCM()
+        # Generate observational data
+        gps_obs, hippocampus_obs, memory_obs = scm.sample(1000)
+        # Encode as a pandas df
+        df = pd.DataFrame(np.vstack([gps_obs, hippocampus_obs, memory_obs]).T, columns=["X", "Z", "Y"])
+
+        # Create the graph describing the causal structure
+        gml_graph = """
+        graph [
+            directed 1
+            
+            node [
+                id "X" 
+                label "X"
+            ]    
+            node [
+                id "Z"
+                label "Z"
+            ]
+            node [
+                id "Y"
+                label "Y"
+            ]
+            node [
+                id "U"
+                label "U"
+            ]
+            
+            edge [
+                source "X"
+                target "Z"
+            ]
+            edge [
+                source "Z"
+                target "Y"
+            ]
+            edge [
+                source "U"
+                target "X"
+            ]
+            edge [
+                source "U"
+                target "Y"
+            ]
+        ]
+        """
+
+        # With graph
+        model = CausalModel(data=df, treatment="X", outcome="Y", graph=gml_graph)
+
+        # model.view_model()
+
+        estimand = model.identify_effect()
+        print(estimand)
+
+        estimate = model.estimate_effect(identified_estimand=estimand, method_name="frontdoor.two_stage_regression")
+        print(f"Estimate of causal effect (linear regression): {estimate.value}")
+
+        refute_subset = model.refute_estimate(
+            estimand=estimand, estimate=estimate, method_name="data_subset_refuter", subset_fraction=0.4
+        )
+        print(refute_subset)
```

Comment:
instead of print, it may be good add an assert, `assert refute_subset is not None`